### PR TITLE
Rough outline of Package Setup

### PR DESCRIPTION
**Pull request by cwickham**
Following general structure of @mbonsma for this, with one exception, I added "Exercise:" where I thought a core exercise might appear.

### CODE REVIEW

File: chapters/03-package-setup.Rmd [hunk 1]
```diff
@@ -20,6 +28,70 @@ collection of classic novels that are freely available from Project Gutenberg,
...
+    * Text: Advantages over sharing code and/or data files:
+        * Easy for other people to obtain and install
+        * Built-in documentation of functions and data
+        * Incorporate package-level meta data: authors, version, etc.
+
+* What is an R package?
+
+    * Text: Files that follow the R package conventions. usethis and devtools, 
+    designed to help you do this.  
+    * Exercise: Install a simple package from GitHub.
```
Comment (mbonsma) [inline chapters/03-package-setup.Rmd:45]: I love this exercise idea!

File: chapters/03-package-setup.Rmd [hunk 2]
```diff
@@ -20,6 +28,70 @@ collection of classic novels that are freely available from Project Gutenberg,
...
+    * Code: `devtools::install()`, `library()`
+    
+* Adding data to a package
+
+    * Text: Data live in `data/` as `.RData` files.  
+    * Code: `usethis::use_data("FIXME")` Here with simplified data. Restart, `devtools::load_all()`, `FIXME`
+    * Text: Should document how data was created. 
+    Put raw data and scripts that end in `usethis::use_data()` in `data-raw/`
+    * Code: `use_data_raw()` 
+    * Exercise: Provide script for getting dracula text from Gutenberg.
```
Comment (mbonsma) [inline chapters/03-package-setup.Rmd:83]: Is this for the reader to do? i.e. if they get stuck, is this a load-bearing step for the package development?

**Review by mbonsma — APPROVED** [review]
This all looks great to me, I can picture this chapter well with this outline!

File: chapters/03-package-setup.Rmd (hunk 2 above)
Comment (cwickham) [reply chapters/03-package-setup.Rmd:83]: I mean we provide the script, the learner has to put it in the right location, end it with `use_data("dracula")` and run it.

Comment (mbonsma) [reply chapters/03-package-setup.Rmd:83]: Ok, sounds great. I don't think it's reasonable or fun to make the exercises only non-essential things, so I guess I was asking a more general question which you also answered!

File: chapters/03-package-setup.Rmd [hunk 3]
```diff
@@ -20,6 +28,70 @@ collection of classic novels that are freely available from Project Gutenberg,
...
+    Give some code that: loads package, 
+    calls `xxx()` and data `yyy`.  
+    Examine structure of GitHub repo.   
+    Can you find where is the `xxx()` defined?  `yyy`? 
+    Where is the name of the package? 
+    
+* Creating a package    
+
+    * Text: Where should your package live? What should you call your package?
+    * Code: `usethis::create_package()`
```
Comment (lwjohnst86) [inline chapters/03-package-setup.Rmd:55]:
```suggestion
    * Code: `usethis::create_package()` (maybe also `available::available()`?)
```

File: chapters/03-package-setup.Rmd [hunk 4]
```diff
@@ -20,6 +28,70 @@ collection of classic novels that are freely available from Project Gutenberg,
...
+    * Code: `devtools::load_all()`
+    * Exercise: Practice workflow. Make small edit to function, reload, 
+    rerun function.
+    * Text: `load_all()` emulates installing and loading, but only
+    for current session.  Sometimes you want to actually install and load.
+    * Code: `devtools::install()`, `library()`
+    
+* Adding data to a package
+
+    * Text: Data live in `data/` as `.RData` files.
```
Comment (lwjohnst86) [inline chapters/03-package-setup.Rmd:78]:
```suggestion
    * Text: Data live in `data/` as `.rda` files.  
```

**Review by lwjohnst86 — APPROVED** [review]
Looks really good so far!!

**Pull request merged**